src/Engine/GameEngine.cs

```diff
@@ -81,5 +88,55 @@ public void OnError(Exception error)
         {
             throw new NotImplementedException();
         }
+
+        #region IDisposable Implementation
+
+        private bool _disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
```

Comment:
To conform with the existing code we avoid using `{ }` on `if`, `for`, etc. If you require a code block, refactor the code block to a well described method. (This is the rule of thumb we go by, but as with all rules, justifiable exceptions can be made).